sksports/model/tests/test_heart_rate.py

```diff
@@ -0,0 +1,75 @@
+"""Tests the module used to estimate the heart-rate."""
+
+# Authors: Aart Goossens
+#          Guillaume Lemaitre
+# License: MIT
+
+import pytest
+import numpy as np
+import pandas as pd
+
+from sksports.model import exp_heart_rate_model
+from sksports.model import HeartRateRegressor
+
+# POWER = np.array([100] * 10 + [240] * 100 + [160] * 100)
+# HEART_RATE = exp_heart_rate_model(POWER, 100, 180, 0.40, 3e-5, 20, 35)
+# HEART_RATE = HEART_RATE + np.random.random(HEART_RATE.size)
+
+
+@pytest.fixture()
+def power_arr():
+    return np.array([100] * 10 + [240] * 100 + [160] * 100)
+
+
+@pytest.fixture(params=[
+    (np.asarray, None),
+    (pd.DataFrame, None),
+    (pd.DataFrame, pd.date_range('1/1/2011', periods=power_arr().size,
+                                 freq='s')),
+    (pd.DataFrame, pd.timedelta_range(start='0 day', periods=power_arr().size,
+                                      freq='s'))])
+def power(request, power_arr):
+    power = request.param[0](power_arr.reshape(-1, 1))
+    if request.param[1] is not None:
+        power.index = request.param[1]
+    return power
+
+
+@pytest.fixture(params=[
```

Comment:
This looks interesting! If I'm correct you're not parametrizing the test but the fixture?...

Reply:
Yes.I would think that fixture are good for data (parametrizing allows to build different supported type. Then parametrizing the test is useful for passing some parameters to the function or something specific to the test. This is just my thumb rule and not sure this is the best ;)